src/meltano/core/plugin/singer/catalog.py

```diff
@@ -247,7 +247,7 @@ def path_property(path: str) -> str:
       stream[0].properties.list_items.properties.account → list_items.account
       stream[0].properties.name                          → name
     """
-    prop_regex = r"properties\.([\w\[\]\d]+)+"
+    prop_regex = r"properties\.([^.]+)+"
```

Comment:
Is it okay that this also matches non-space whitespace characters such as newlines?

Reply:
Yeah, I think it's very rare that a metadata/schema path expression would contain a newline or similar characters but this should keep